autometa/common/external/diamond.py

```diff
@@ -231,28 +347,33 @@ def blast(
         "6",
         "--out",
         outfpath,
-        "--tmpdir",
-        tmpdir,
     ]
+    if tmpdir:
+        cmd.extend(["--tmpdir", tmpdir])
+    # this is done as when cmd is a list each element should be a string
     cmd = [str(c) for c in cmd]
     if verbose:
         logger.debug(f'RunningDiamond: {" ".join(cmd)}')
-    with open(os.devnull, "w") as stdout, open(os.devnull, "w") as stderr:
-        proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
-    if proc.returncode:
-        raise OSError(f"DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}")
+    subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True
+    )
     return outfpath
 
 
-def parse(results, top_pct=0.9, verbose=False):
-    """Retrieve diamond results from output table
+def parse(results, top_percentile=0.9, verbose=False):
```

Comment:
You ignored my request in issue #36. Please read it again:

>In the script, there is an argument/variable called top_pct or a variant of this name. This is a float value between 0 and 1 (although it can't exactly be zero), used to determine if the bitscore of hits is above a threshold. For example, if it is 0.9 then only bitscores >= 0.9 * the top bitscore are accepted. The problem here is not one of programmatic logic, but rather readability for contributors. top_pct looks like "top percentage", which would suggest, for example, that for a value of 0.9, 90% will be kept. However, the opposite is true in this case - only 10% will be kept. This is compounded by the help message in the if __name__ == '__main__': block, which describes this argument as "top percentage of hits to retrieve". So what I am saying is that the variable name and documentation is confusing. Perhaps it could be called "bitscore_tolerance" or something? Accompanied by a change in documentation and inclusion of explanatory comments.

The variable does not describe a "top percentile" to keep, but rather some sort of threshold.

Reply:
I was thinking of`bitscr_filter` with the corresponding `help` text:
"hits with bitscore greater than bitscr-filter*top bitscore will be kept".
Other options that I thought were `bitscore_threshold` and the one you suggested i.e. `bitscr_tolerance`. I was moe inclined towards `bitscr_filter`.
What do you think about it? Any ideas?

Reply:
`bitscore_filter` is OK, but can you not abbreviate? If you have a decent text editor it should autocomplete anyway.

Reply:
Alright!
sure I can write the full name. I don't know why I was of the view that developers prefer variable names that have less character (i.e. abbreviated), seems like I was wrong. I'll make the change and push the changes!